From: Reiner Herrmann <reiner@reiner-h.de>
Date: Sat, 2 Sep 2017 09:55:59 -0300
Subject: lib/mkmf.rb: sort list of object files in generated Makefile

Without sorting the list explicitely, its order is indeterministic,
because readdir() is also not deterministic.
When the list of object files varies between builds, they are linked
in a different order, which results in an unreproducible build.

Signed-off-by: Reiner Herrmann <reiner@reiner-h.de>
Signed-off-by: Antonio Terceiro <terceiro@debian.org>
---
 lib/mkmf.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index 3181b05..2502476 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -2275,7 +2275,7 @@ def create_makefile(target, srcprefix = nil)
 LIBS = #{$LIBRUBYARG} #{$libs} #{$LIBS}
 ORIG_SRCS = #{orig_srcs.collect(&File.method(:basename)).join(' ')}
 SRCS = $(ORIG_SRCS) #{(srcs - orig_srcs).collect(&File.method(:basename)).join(' ')}
-OBJS = #{$objs.join(" ")}
+OBJS = #{$objs.sort.join(" ")}
 HDRS = #{hdrs.map{|h| '$(srcdir)/' + File.basename(h)}.join(' ')}
 TARGET = #{target}
 TARGET_NAME = #{target && target[/\A\w+/]}
